testing/sdk_install.py

```diff
@@ -52,30 +77,30 @@ def install(
         timeout_seconds=TIMEOUT_SECONDS,
         wait_for_deployment=True):
     start = time.time()
-    merged_options = get_package_options(additional_options)
 
-    log.info('Installing {}:{} with options={} version={}'.format(
-        package_name, service_name, merged_options, package_version))
+    # If the package is already installed at this point, fail immediately.
+    if sdk_marathon.app_exists(service_name):
+        raise dcos.errors.DCOSException('Service is already installed: {}'.format(service_name))
 
     # 1. Install package, wait for tasks, wait for marathon deployment
-    retried_shakedown_install(
+    _retried_install_impl(
         package_name,
         service_name,
+        expected_running_tasks,
+        get_package_options(additional_options),
         package_version,
-        merged_options,
-        timeout_seconds,
-        expected_running_tasks)
+        timeout_seconds)
 
     # 2. Wait for the scheduler to be idle (as implied by deploy plan completion and suppressed bit)
     # This should be skipped ONLY when it's known that the scheduler will be stuck in an incomplete state.
     if wait_for_deployment:
         # this can take a while, default is 15 minutes. for example with HDFS, we can hit the expected
         # total task count via FINISHED tasks, without actually completing deployment
-        log.info("Waiting for {}/{} to finish deployment plan...".format(
+        log.info('Waiting for package={} service={} to finish deployment plan...'.format(
```

Comment:
Since we're editing this line anyway, we could switch to using `%s` in log messages instead of `{}` and `format` to keep linters happy.

Reply:
```
dcos-commons$ grep -R "{}" testing/ frameworks/*/tests/ | wc -l
499
dcos-commons$ egrep -R "%[sd]" testing/ frameworks/*/tests/ | wc -l
63
```

Reply:
Yes, I'm not saying that we're consistent.